include/pre-charge/pre_charge.hpp

```diff
@@ -0,0 +1,202 @@
+     * Get the state of MC_KEY_IN
+     * 
+     */
+    void getMCKey();
+
+    /**
+     * Set the Precharge state
+     * 
+     * @param state 0 = precharge disabled, 1 = precharge enabled
+     */
+    void setPrecharge(int state);
+
+    /**
+     * Set the Discharge state
+     * 
+     * @param state 0 = discharge disabled, 1 = discharge enabled
+     */
+    void setDischarge(int state);
+
+    /**
+     * Toggle the state of the Main Contactor
+     * 
+     * @param state 0 = contactor open, 1 = contactor closed
+     */
+    void setContactor(int state);
+
+    /**
+     * Set the Forward state
+     * 
+     * @param state 0 = forward disabled, 1 = forward enabled
+     */
+    void setForward(int state);
+
+    /**
+     * Set the APM state
+     * 
+     * @param state 0 = APM disabled, 1 = APM enabled
+     */
+    void setAPM(int state);
+
+    /**
+     * Handles when the MC is powered off.
+     * 
+     * State: State::MC_OFF
+     */
+    void mcOffState();
+
+    /**
+     * Handles when the MC is powered on.
+     * 
+     * State: State::MC_ON
+     */
+    void mcOnState();
+
+    /**
+     * Handles when an unsafe condition is detected while the MC is off.
+     * 
+     * State: State::ESTOPWAIT
+     */
+    void eStopState();
+
+    /**
+     * Handles when precharge is to occur.
+     * 
+     * State: State::PRECHARGE
+     */
+    void prechargeState();
+
+    /**
+     * Handles when discharge is to occur.
+     * 
+     * State: State::DISCHARGE
+     */
+    void dischargeState();
+
+    /**
+     * Handles when the contactor is to be open.
+     * 
+     * State: State::CONT_OPEN
+     */
+    void contOpenState();
+
+    /**
+     * Handles when the contactor is to be closed.
+     * 
+     * State: State::CONT_CLOSE
+     */
+    void contCloseState();
+
+    /**
+     * Handles when the Forward Enable is to be disabled.
+     * 
+     * State: State::FORWARD_DISABLE
+     */
+    void forwardDisableState();
+
+    /**
+     * Pretty print the current state machine status for testing
+     * 
+     */
+    std::string printState();
+
+    /**
+     * Testing helper function to return the string value of
+     * the current state enum
+     * 
+     * @param currentState 
+     * @return std::string 
+     */
+    std::string stateString(State currentState);
+
+private:
+    /** GPIO instance to monitor KEY_IN */
+    IO::GPIO& key;
+    /** GPIO instance to monitor BATTERY_1_OK */
+    IO::GPIO& batteryOne;
+    /** GPIO instance to monitor BATTERY_2_OK */
+    IO::GPIO& batteryTwo;
+    /** GPIO instance to monitor ESTOP_STATUS */
+    IO::GPIO& eStop;
+    /** GPIO instance to toggle PC_CTL */
+    IO::GPIO& pc;
+    /** GPIO instance to toggle DC_CTL */
+    IO::GPIO& dc;
+    /** GPIO instance to toggle CONT_CTL */
+    IO::GPIO& cont;
+    /** GPIO instance to toggle APM_CTL */
+    IO::GPIO& apm;
+    /** GPIO instance to toggle FW_EN_CTL */
+    IO::GPIO& forward;
+
+    IO::GPIO::State keyStatus;
```

Comment:
NIT: Should probably have a different enum to represent these states since its not as semantically meaningful to say the `batteryOneStatus` is `HIGH`. Is `HIGH` good? bad?